packages/react/src/components/date-picker/calendar-header.tsx

```diff
@@ -100,8 +100,8 @@ export const CalendarHeader: VoidFunctionComponent<CalendarHeaderProps> = ({
                         ariaLabel={t('yearSelectLabel')}
                         data-testid="year-select"
                         options={yearsOptions}
-                        onChange={(options) => {
-                            changeYear(parseInt(options.value, 10));
+                        onChange={(options: DropdownListOption) => {
+                            changeYear(parseInt(options?.value, 10));
```

Comment:
`options` n'est pas nullable
```suggestion
                            changeYear(parseInt(options.value, 10));
```